src/main/scala/mesosphere/marathon/state/PathId.scala

```diff
@@ -137,12 +137,21 @@ object PathId {
     id.path.forall(part => ID_PATH_SEGMENT_PATTERN.pattern.matcher(part).matches())
   }
 
+  private val reservedKeywordsSeq = Seq("restart", "tasks", "versions")
+
+  private val reservedKeywords = reservedKeywordsSeq.toSet
+
+  private val withoutReservedKeywords = isTrue[PathId](s"must not end with any of the following reserved keywords: ${reservedKeywordsSeq.mkString(", ")}") { id =>
```

Comment:
I do understand the reason of having both a `Seq` and a `Set`, but it feels a bit weird on having a copy of the same data twice. Perhaps do something like this to achieve order consistency on the error messages:

```scala
private val reservedKeywordsError = s"must not end with any of the following reserved keywords: ${reservedKeywords.toSeq.sorted.mkString(", ")}"
```

Reply:
I think the idea was to have an efficient lookup. I'd say premature optimization here :) Let's just use the Seq and drop the Set.